File: BEMCheckBox/iOS Simulator/BEMCheckBox.cs
// Import of BEMCheckBox ()
// Frameworks: BEMCheckBox
// Targets: i386, x86_64
// Dep fx:rtl, UIKit, Foundation, QuartzCore, CoreGraphics
// Dep libs:
// Platform: macOS
// 


namespace
{
	class BEMCheckBox.BEMCheckBox.BEMCheckBoxGroup : NSObject
	{
		public NSHashTable! checkBoxes { get; set; }
		public BEMCheckBox.BEMCheckBox.BEMCheckBox? selectedCheckBox { get; set; }
		public BOOL mustHaveSelection { get; set; }
		[NonSwiftOnly]
		public static instancetype! groupWithCheckBoxes(NSArray<BEMCheckBox.BEMCheckBox.BEMCheckBox> checkBoxes);
		[Alias]
		[SwiftOnly]
		public static instancetype! @group(NSArray<BEMCheckBox.BEMCheckBox.BEMCheckBox> checkBoxes);
		[InitFromClassFactoryMethod]
		[Alias]
		[SwiftOnly]
		public static this withCheckBoxes(NSArray<BEMCheckBox.BEMCheckBox.BEMCheckBox> checkBoxes);
		[NonSwiftOnly]
		public void addCheckBoxToGroup(BEMCheckBox.BEMCheckBox.BEMCheckBox! checkBox);
		[Alias]
		[SwiftOnly]
		public void addCheckBox(BEMCheckBox.BEMCheckBox.BEMCheckBox! checkBox);
		[NonSwiftOnly]
		public void removeCheckBoxFromGroup(BEMCheckBox.BEMCheckBox.BEMCheckBox! checkBox);
		[Alias]
		[SwiftOnly]
		public void removeCheckBox(BEMCheckBox.BEMCheckBox.BEMCheckBox! checkBox);
	}

	enum BEMCheckBox.BEMCheckBox.BEMBoxType
	{
		[NonSwiftOnly] BEMBoxTypeCircle = 0,
		[NonSwiftOnly] Circle = 0,
		[SwiftOnly] circle = 0,
		[NonSwiftOnly] BEMBoxTypeSquare = 1,
		[NonSwiftOnly] Square = 1,
		[SwiftOnly] square = 1
	}

	class BEMCheckBox.BEMCheckBox.BEMCheckBox : UIControl, ICAAnimationDelegate
	{
		public BEMCheckBox.BEMCheckBox.IBEMCheckBoxDelegate! @delegate { get; set; }
		public BOOL @on { get; set; }
		public CGFloat lineWidth { get; set; }
		public CGFloat cornerRadius { get; set; }
		public CGFloat animationDuration { get; set; }
		public BOOL hideBox { get; set; }
		public UIColor! onTintColor { get; set; }
		public UIColor! onFillColor { get; set; }
		public UIColor! offFillColor { get; set; }
		public UIColor! onCheckColor { get; set; }
		public UIColor! tintColor { get; set; }
		public BEMCheckBox.BEMCheckBox.BEMCheckBoxGroup? @group { get; set; }
		public BEMCheckBox.BEMCheckBox.BEMBoxType boxType { get; set; }
		public BEMAnimationType onAnimationType { get; set; }
		public BEMAnimationType offAnimationType { get; set; }
		public CGSize minimumTouchSize { get; set; }
		public void setOn(BOOL @on) animated(BOOL animated);
		public void reload();
		public UIColor! onFill { get; set; }
		public UIColor! offFill { get; set; }
		public UIColor! onCheck { get; set; }
		public CGSize minimumTouch { get; set; }
	}

	using BEMAnimationType = BEMCheckBox.BEMCheckBox.__enum_BEMAnimationType;

	enum BEMCheckBox.BEMCheckBox.__enum_BEMAnimationType
	{
		[NonSwiftOnly] BEMAnimationTypeStroke = 0,
		[NonSwiftOnly] Stroke = 0,
		[SwiftOnly] stroke = 0,
		[NonSwiftOnly] BEMAnimationTypeFill = 1,
		[NonSwiftOnly] Fill = 1,
		[SwiftOnly] fill = 1,
		[NonSwiftOnly] BEMAnimationTypeBounce = 2,
		[NonSwiftOnly] Bounce = 2,
		[SwiftOnly] bounce = 2,
		[NonSwiftOnly] BEMAnimationTypeFlat = 3,
		[NonSwiftOnly] Flat = 3,
		[SwiftOnly] flat = 3,
		[NonSwiftOnly] BEMAnimationTypeOneStroke = 4,
		[NonSwiftOnly] OneStroke = 4,
		[SwiftOnly] oneStroke = 4,
		[NonSwiftOnly] BEMAnimationTypeFade = 5,
		[NonSwiftOnly] Fade = 5,
		[SwiftOnly] fade = 5
	}

	interface BEMCheckBox.BEMCheckBox.IBEMCheckBoxDelegate : INSObject
	{
		[NonSwiftOnly]
		void didTapCheckBox(BEMCheckBox.BEMCheckBox.BEMCheckBox! checkBox);
		[Alias]
		[SwiftOnly]
		void didTap(BEMCheckBox.BEMCheckBox.BEMCheckBox! checkBox);
		[NonSwiftOnly]
		void animationDidStopForCheckBox(BEMCheckBox.BEMCheckBox.BEMCheckBox! checkBox);
		[Alias]
		[SwiftOnly]
		void animationDidStop(BEMCheckBox.BEMCheckBox.BEMCheckBox! checkBox);
	}

	using BEMCheckBoxDelegate = BEMCheckBox.BEMCheckBox.IBEMCheckBoxDelegate;

}

File: BEMCheckBox/iOS Simulator/BEMCheckBox.swift
// Import of BEMCheckBox ()
// Frameworks: BEMCheckBox
// Targets: i386, x86_64
// Dep fx:rtl, UIKit, Foundation, QuartzCore, CoreGraphics
// Dep libs:
// Platform: macOS
// 

class BEMCheckBox.BEMCheckBox.BEMCheckBoxGroup : NSObject { 
	public var checkBoxes: NSHashTable
	public var selectedCheckBox: BEMCheckBox.BEMCheckBox.BEMCheckBox?
	public var mustHaveSelection: BOOL!
	@NonSwiftOnly
	public static func groupWithCheckBoxes(_ checkBoxes: NSArray<BEMCheckBox.BEMCheckBox.BEMCheckBox!>!) -> instancetype
	@Alias
	@SwiftOnly
	public static func group(checkBoxes: NSArray<BEMCheckBox.BEMCheckBox.BEMCheckBox!>!) -> instancetype
	@InitFromClassFactoryMethod
	@Alias
	@SwiftOnly
	public init(checkBoxes: NSArray<BEMCheckBox.BEMCheckBox.BEMCheckBox!>!)
	@NonSwiftOnly
	public func addCheckBoxToGroup(_ checkBox: BEMCheckBox.BEMCheckBox.BEMCheckBox)
	@Alias
	@SwiftOnly
	public func addCheckBox(toGroup checkBox: BEMCheckBox.BEMCheckBox.BEMCheckBox)
	@NonSwiftOnly
	public func removeCheckBoxFromGroup(_ checkBox: BEMCheckBox.BEMCheckBox.BEMCheckBox)
	@Alias
	@SwiftOnly
	public func removeCheckBox(fromGroup checkBox: BEMCheckBox.BEMCheckBox.BEMCheckBox)
}

enum BEMCheckBox.BEMCheckBox.BEMBoxType { 
	@NonSwiftOnly
	case BEMBoxTypeCircle = 0
	@NonSwiftOnly
	case Circle = 0
	@SwiftOnly
	case circle = 0
	@NonSwiftOnly
	case BEMBoxTypeSquare = 1
	@NonSwiftOnly
	case Square = 1
	@SwiftOnly
	case square = 1
}

class BEMCheckBox.BEMCheckBox.BEMCheckBox : UIControl, ICAAnimationDelegate { 
	public var delegate: BEMCheckBox.BEMCheckBox.IBEMCheckBoxDelegate
	public var on: BOOL!
	public var lineWidth: CGFloat!
	public var cornerRadius: CGFloat!
	public var animationDuration: CGFloat!
	public var hideBox: BOOL!
	public var onTintColor: UIColor
	public var onFillColor: UIColor
	public var offFillColor: UIColor
	public var onCheckColor: UIColor
	public var tintColor: UIColor
	public var group: BEMCheckBox.BEMCheckBox.BEMCheckBoxGroup?
	public var boxType: BEMCheckBox.BEMCheckBox.BEMBoxType!
	public var onAnimationType: BEMAnimationType!
	public var offAnimationType: BEMAnimationType!
	public var minimumTouchSize: CGSize!
	public func setOn(_ on: BOOL!, animated: BOOL!)
	public func reload()
	public var onFill: UIColor
	public var offFill: UIColor
	public var onCheck: UIColor
	public var minimumTouch: CGSize!
}

typealias BEMAnimationType = BEMCheckBox.BEMCheckBox.__enum_BEMAnimationType!

enum BEMCheckBox.BEMCheckBox.__enum_BEMAnimationType { 
	@NonSwiftOnly
	case BEMAnimationTypeStroke = 0
	@NonSwiftOnly
	case Stroke = 0
	@SwiftOnly
	case stroke = 0
	@NonSwiftOnly
	case BEMAnimationTypeFill = 1
	@NonSwiftOnly
	case Fill = 1
	@SwiftOnly
	case fill = 1
	@NonSwiftOnly
	case BEMAnimationTypeBounce = 2
	@NonSwiftOnly
	case Bounce = 2
	@SwiftOnly
	case bounce = 2
	@NonSwiftOnly
	case BEMAnimationTypeFlat = 3
	@NonSwiftOnly
	case Flat = 3
	@SwiftOnly
	case flat = 3
	@NonSwiftOnly
	case BEMAnimationTypeOneStroke = 4
	@NonSwiftOnly
	case OneStroke = 4
	@SwiftOnly
	case oneStroke = 4
	@NonSwiftOnly
	case BEMAnimationTypeFade = 5
	@NonSwiftOnly
	case Fade = 5
	@SwiftOnly
	case fade = 5
}

protocol BEMCheckBox.BEMCheckBox.IBEMCheckBoxDelegate : INSObject { 
	@NonSwiftOnly
	func didTapCheckBox(_ checkBox: BEMCheckBox.BEMCheckBox.BEMCheckBox)
	@Alias
	@SwiftOnly
	func didTap(_ checkBox: BEMCheckBox.BEMCheckBox.BEMCheckBox)
	@NonSwiftOnly
	func animationDidStopForCheckBox(_ checkBox: BEMCheckBox.BEMCheckBox.BEMCheckBox)
	@Alias
	@SwiftOnly
	func animationDidStop(`for` checkBox: BEMCheckBox.BEMCheckBox.BEMCheckBox)
}

typealias BEMCheckBoxDelegate = BEMCheckBox.BEMCheckBox.IBEMCheckBoxDelegate!

